weave/ops_arrow/convert.py

```diff
@@ -289,7 +294,10 @@ def to_arrow_from_list_and_artifact(
 
 
 def to_arrow(
-    obj, wb_type=None, artifact: typing.Optional[artifact_base.Artifact] = None
+    obj,
+    wb_type=None,
+    artifact: typing.Optional[artifact_base.Artifact] = None,
+    py_objs_already_mapped: bool = False,
```

Comment:
hmm... this feels like a smell. why would we need to call `to_arrow` if the objects are already mapped?

Reply:
hmmm ... i see your point. This relates to the above comments. Upon reflection, maybe it is better to do the _val stuff externally, then directly construct the awl. I'll revisit this today

Reply:
hmmm unions are sort of nasty here... we still want the mappers to handle this